provider/service.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2016 IBM Corp. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from consumer import Consumer
+from database import Database
+from threading import Thread
+
+class Service (Thread):
+    def __init__(self, consumers):
+        Thread.__init__(self)
+        self.daemon = True
+
+        self.changes = Database().changesFeed()
+        self.consumers = consumers
+
+    def run(self):
+        while True:
+            for change in self.changes:
+                if "deleted" in change and change["deleted"] == True:
+                    logging.info('[changes] Found a delete')
+                    consumer = self.consumers.getConsumerForTrigger(change['id'])
+                    if consumer != None:
+                        consumer.shutdown()
+
+                else:
+                    logging.info('[changes] Found a change')
+                    if not self.consumers.hasConsumerForTrigger(change["id"]):
+                        # since we can't use a filter function for the feed (then
+                        # you don't get deletes) we need to manually verify this
+                        # is a valid trigger doc that has changed
+                        if 'triggerURL' in change['doc']:
+                            self.createAndRunConsumer(change['doc'])
+                        else:
+                            logging.debug('[changes] Found a change for a non-trigger document')
+                    else:
+                        # this isn't really supported just yet...
+                        logging.warn("[{}] Found a document update to an existing trigger... that's unexpected".format(change['id']))
+
+            logging.error("[changes] uh-oh! I made it out of the changes for loop!")
```

Comment:
Because the changes feed is "infinite" the `for` loop will keep going indefinitely even when there are no changes at the moment. Which is to say, line 52 is only ever executed when something unexpected happens. That's why it is logged as an error. The outer `while True` loop will automatically restart checking the changes feed.